tests/std/tests/GH_002039_byte_is_not_trivially_swappable/test.compile.pass.cpp

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <chrono>
+#include <cstddef>
+#include <type_traits>
+
+using namespace std;
+
+int main() {} // COMPILE-ONLY
+
+#define STATIC_ASSERT(...) static_assert(__VA_ARGS__, #__VA_ARGS__)
+
+#ifdef __cpp_lib_byte
+STATIC_ASSERT(_Is_trivially_swappable_v<byte>);
+#endif // __cpp_lib_byte
+STATIC_ASSERT(_Is_trivially_swappable_v<unsigned char>);
+STATIC_ASSERT(_Is_trivially_swappable_v<unsigned short>);
+STATIC_ASSERT(_Is_trivially_swappable_v<unsigned int>);
+STATIC_ASSERT(_Is_trivially_swappable_v<unsigned long>);
+STATIC_ASSERT(_Is_trivially_swappable_v<unsigned long long>);
+
+STATIC_ASSERT(_Is_trivially_swappable_v<signed char>);
+STATIC_ASSERT(_Is_trivially_swappable_v<short>);
+STATIC_ASSERT(_Is_trivially_swappable_v<int>);
+STATIC_ASSERT(_Is_trivially_swappable_v<long>);
+STATIC_ASSERT(_Is_trivially_swappable_v<long long>);
+
+STATIC_ASSERT(_Is_trivially_swappable_v<char>);
+STATIC_ASSERT(_Is_trivially_swappable_v<wchar_t>);
+
+STATIC_ASSERT(_Is_trivially_swappable_v<float>);
+STATIC_ASSERT(_Is_trivially_swappable_v<double>);
+STATIC_ASSERT(_Is_trivially_swappable_v<long double>);
+
+enum class byte2 : unsigned char {};
+STATIC_ASSERT(_Is_trivially_swappable_v<byte2>);
+
+enum class byte3 : unsigned char {};
+
+void swap(byte3& a, byte3& b) {
+    byte3 temp{a};
+    a = b;
+    b = temp;
+}

```

Comment:
It's conventional to declare but not define functions in COMPILE-ONLY tests - function bodies are just noise. I would
 ```suggestion
void swap(byte3&, byte3&);
```
here, and similarly avoid defining the non-trivial member functions in the classes below.